prometheus-v2ray-exporter: init at 0.6.0

jqqqqqqqqqq opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
I have been using this exporter to on my nixos servers for half a year and it is very stable. Wanna share it to everyone and make my local nixpkgs identical with upstream.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [X] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

r-rmcgibbo commented:
Result of `nixpkgs-review pr 130149` at 29d37c52 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>prometheus-v2ray-exporter</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `stdenv`.
Near pkgs/servers/monitoring/prometheus/v2ray-exporter.nix:1:8:
```
  |
1 | { lib, stdenv, buildGoModule, fetchFromGitHub, nixosTests }:
  |        ^
```
</li>
    <li>warning: unused-argument

Unused argument: `nixosTests`.
Near pkgs/servers/monitoring/prometheus/v2ray-exporter.nix:1:48:
```
  |
1 | { lib, stdenv, buildGoModule, fetchFromGitHub, nixosTests }:
  |                                                ^
```
</li>
  </ul>
</details>

Ma27 reviewed (commented):
The change itself looks fine to me, but a VM-test would be appreciated :)

jqqqqqqqqqq commented:
Any update? I probably don't have time to write a VM-test... Will do when I update this package again

jqqqqqqqqqq commented:
@Ma27 Hi, I have added the unit tests and also updated the package version. Should I clean the commits up and create a new PR? Also it seems this unit test requires some approval which I don't know how to pass.

jqqqqqqqqqq commented:
@WilliButz Hi! Any chance to take a look? It's been so long since this pr is created. Thank you!

Ma27 commented:
Are you sure you pushed the VM tests to this branch? Also, yes, please clean up the history. No need to create a new PR though.

jqqqqqqqqqq commented:
I probably did something wrong in my repo and this pr is shown as merged...
Let me open a new one with same content.

jqqqqqqqqqq commented:
@Ma27 I have opened a new pr because my rebase corrupted the status of this pr. Can you take a look? https://github.com/NixOS/nixpkgs/pull/156813
